### PR TITLE
[jit] Fix the devirt of EqualityComparer<byte>.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by lateralusX** [comment]
Fix assert seen on full AOT targets using code like this:

```
public struct DymmyValueType
{
    byte dymmy_value;
}

static void Main ()
{
    var someData = new Dictionary<byte, DymmyValueType>();
    someData[1] = new DymmyValueType();
    someData[1] = new DymmyValueType();
}

```

File: mono/mini/aot-tests.cs [hunk 1]
```diff
@@ -669,6 +669,17 @@ class GetCurrentMethodClass<T> {
 		return 0;
 	}
 
+	struct ByteStruct {
+		byte dymmy_value;
+	}
+
+	public static int test_0_byte_equality_compater_devirt () {
+		var dict = new Dictionary<byte, AStruct>();
+		dict [1] = new AStruct ();
```
Comment (lateralusX) [inline mono/mini/aot-tests.cs]: ByteStruct ?

File: mono/mini/aot-tests.cs [hunk 2]
```diff
@@ -669,6 +669,17 @@ class GetCurrentMethodClass<T> {
...
+		dict [1] = new AStruct ();
```
Comment (lateralusX) [inline mono/mini/aot-tests.cs]: ByteStruct ?

File: mono/mini/aot-tests.cs (hunk 1 above)
Comment (lateralusX) [reply mono/mini/aot-tests.cs]: or reuse of other struct, then no need for ByteStruct.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**